articles/feature-engineering-in-machine-learning/index.md

```diff
@@ -0,0 +1,167 @@
+
+- **Regression models** – unlike classification models, these tasks give output variables that take continuous values. An example of such a model is a linear regression model.
+
+#### Machine Learning Categories
+There are three types of machine learning:
+
+- **Supervised learning** - an example of this is a student being supervised by a teacher. The student is trying to solve a problem in class. The teacher oversees that the student solves the problem correctly. For a model, this happens through use of labeled data. The labeled data, as input, guides the model. Classification and regression models are supervised learning algorithms.
+
+- **Reinforcement learning** - learning is through a trial-and-error approach. An agent/machine performs actions and is either rewarded or learns from errors.
+
+- **Unsupervised learning** - simply the opposite of supervised learning. There is no clean, complete, or labelled dataset provided. While supervised learning output relies on labeled data, unsupervised output is based on the patterns identified in input data. Input data has no pre-existing labels.
+
+We are now acquainted with machine learning. But, in order to understand what feature engineering is, we need to have an overview of the entire machine learning process.
+
+#### Machine Learning Process
+It is important to understand the machine learning lifecycle. These are more or less the steps of the machine learning process:
+
+- **Collecting data**- data is gathered from various sources. In any machine learning workflow, this is the first step. It is a very important step since the right data translates to better results.
+
+- **Preparing that data**- to get some form of meaning out of data, a number of processes take place in this stage. Datasets are evaluated for missing values, outliers, trends, and so on. Oddities in data are rectified. Data is cleaned. It is worth noting that the feature engineering process takes place here. I argue this to be the most important step in the lifecycle.
+
+- **Choosing a model**- this step involves selection of a model best suited to the prepared data.
+
+- **Training**-  the model is trained in this stage (using data). Data may be split as test data and training data. Test data is used to test the model. Training data trains the model.
+
+- **Evaluation**- the model performance is evaluated at this stage through a number of metrics.
```

Comment:
Define the metrics for a better understanding of the article. Name a few of the evaluation metrics.